wdns/str_to_rdata_ubuf.c

```diff
@@ -218,9 +218,9 @@ str_to_svcparam(ubuf *u, char *keyval)
 		}
 
 		tmpul = strtoul(tok, &endport, 10);
+		assert(endport != NULL);
```

Comment:
This assertion isn't needed, nor is the corresponding one in svcparamkeys_to_str.c.

`strtoul`, by [definition](https://pubs.opengroup.org/onlinepubs/9699919799/functions/strtoul.html), stores a pointer to a portion of the string (the "final string" in the linked specification) in the location pointed to by its second argument. Thus, we can rely on `strtoul` populating `endport` with a non-NULL value here.

Reply:
Ok